frontend/lib/src/components/elements/ArrowVegaLiteChart/useVegaEmbed.ts

```diff
@@ -137,7 +140,7 @@ export function useVegaEmbed(
       vegaFinalizer.current = finalize
 
       // Load the initial set of data into the chart.
-      const dataArrays = getDataArrays(datasetsRef.current)
+      const dataArrays = getDataArrays(datasets)
```

Comment:
Under heavy reruns (e.g., many sidebar widgets), these refs can lag and hold the old chart’s datasets, which don’t match the new spec’s dataset names, causing Vega to throw on insert.

I don't think there is a downside to directly using the initial `datasets` here.